pkg/machine/qemu/machine.go

```diff
@@ -680,14 +688,14 @@ func (v *MachineVM) isRunning() (bool, error) {
 	return false, nil
 }
 
-func (v *MachineVM) isListening() bool {
+func (v *MachineVM) isListening() (bool, error) {
 	// Check if we can dial it
 	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", "localhost", v.Port), 10*time.Millisecond)
 	if err != nil {
-		return false
+		return false, nil
```

Comment:
Why aren't you returning err?

Reply:
Currently it loops until it is up and running, if there is a dial timeout that would be an error (and abort sleeping).

But basically I just copied the logic from the isRunning, which also returns `nil` for missing and err for errors

Reply:
Right now it doesn't check protocol (ssh), but it could return an error if it wasn't responding correctly or something ?

Which would indicate that the machine was running and listening, but needs to be restarted before it will "work"

Reply:
Well if the function never returns an a real error, then it shouldn't return an error at all, and  you should add a comment about the nil on error.

Reply:
Maybe it should go back to just returning a boolean.